File: src/bfs/q_1260.java
package bfs;

import java.io.*;
import java.util.*;

public class q_1260 {
	static boolean visited[];
	static ArrayList<Integer>[] A;

	public static void main(String[] args) throws Exception {
		Scanner sc = new Scanner(System.in);
		int N = sc.nextInt();
		int M = sc.nextInt();
		int Start = sc.nextInt();
		A = new ArrayList[N + 1];
		for (int i = 1; i < N + 1; i++) {
			A[i] = new ArrayList<Integer>();
		}
		for (int i = 0; i < M; i++) {
			int S = sc.nextInt();
			int E = sc.nextInt();
			A[S].add(E);
			A[E].add(S);
		}

		for (int i = 1; i <= N; i++) {
			Collections.sort(A[i]);
		}
		visited = new boolean[N + 1];
		DFS(Start);
		System.out.println();
		visited = new boolean[N + 1];
		BFS(Start);
		System.out.println();
	}

	static void DFS(int Node) {
		System.out.print(Node + " ");
		visited[Node] = true;
		for (int i : A[Node]) {
			if (!visited[i]) {
				DFS(i);
			}
		}
	}

	static void BFS(int Node) {
		Queue<Integer> queue = new LinkedList<Integer>();
		queue.add(Node);
		visited[Node] = true;

		while (!queue.isEmpty()) {
			int now_Node = queue.poll();
			System.out.print(now_Node + " ");
			for (int i : A[now_Node]) {
				if (!visited[i]) {
					visited[i] = true;
					queue.add(i);
				}
			}
		}

	}

}

File: kotlin/bfs/q_1260.kt
package bfs

import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.StringTokenizer


lateinit var visited: BooleanArray
lateinit var A: Array<MutableList<Int>>

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val st = StringTokenizer(br.readLine())

    val N = st.nextToken().toInt()
    val M = st.nextToken().toInt()
    val start = st.nextToken().toInt()

    A = Array(N + 1) { mutableListOf() }

    repeat(M) {
        val st2 = StringTokenizer(br.readLine())
        val S = st2.nextToken().toInt()
        val E = st2.nextToken().toInt()
        A[S].add(E)
        A[E].add(S)
    }

    for (i in 1..N) {
        A[i].sort()
    }

    visited = BooleanArray(N + 1)
    dfs(start)
    println()

    visited = BooleanArray(N + 1)
    bfs(start)
    println()
}

fun dfs(node: Int) {
    print("$node ")
    visited[node] = true

    for (i in A[node]) {
        if (!visited[i]) {
            dfs(i)
        }
    }
}

fun bfs(node: Int) {
    val queue = ArrayDeque<Int>()
    queue.add(node)
    visited[node] = true

    while (queue.isNotEmpty()) {
        val now = queue.removeFirst()
        print("$now ")

        for (i in A[now]) {
            if (!visited[i]) {
                visited[i] = true
                queue.addLast(i)
            }
        }
    }
}